From f110a8c2844781b5c5eb39339448de19b2a59ad3 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Mon, 16 Oct 2023 09:23:58 +0200
Subject: [PATCH] Adapt to new quota output

Showing size of the snapshots no longer works. In the past the output of
the quota command looked like this:

qgroupid         rfer         excl
--------         ----         ----
0/5          16.00KiB     16.00KiB

Nowadays it contains a patch as well and the final output is as follows:

Qgroupid    Referenced    Exclusive   Path
--------    ----------    ---------   ----
0/1386         2.46GiB     60.34MiB   @135

So updating schnapps to be able to parse the second output as well. In
theory even better would be to use json output and do the math to get
human readable output, but doing math in shell is not a great idea, so
sticking with the text output for now.
---
 schnapps.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/schnapps.sh b/schnapps.sh
index 9f7253a..caf9353 100755
--- a/schnapps.sh
+++ b/schnapps.sh
@@ -288,6 +288,8 @@ generic_list() {
             DESCRIPTION="$(filter_description "$DESCRIPTION")"
             [ ! -d "$dir/@$snapshot" ] \
                 || SIZE="$(btrfs qgroup show -f "$dir/@$snapshot" | sed -n 's|.*[[:blank:]]\([0-9.MGKi]*B\)[[:blank:]]*$|\1|p')"
+            [ ! -d "$dir/@$snapshot" -a -z "$SIZE" ] \
+                || SIZE="$(btrfs qgroup show -f "$dir/@$snapshot" | sed -n 's|.*[[:blank:]]\([0-9.MGKi]*B\)[[:blank:]]*'"@$snapshot"'[[:blank:]]*$|\1|p')"
             [ ! -f "$dir/$snapshot".tar.gz ] \
                 || SIZE="$(du -sh "$dir/$snapshot".tar.gz | sed 's|[[:blank:]].*||')"
             [ ! -f "$dir/$snapshot".tar.gz.pgp ] \
-- 
2.42.0

